llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -447,6 +467,24 @@ static string_vector saveDeviceImageProperty(
       }
     }
 
+    // Metadata names may be composite so we keep them alive until the
+    // properties have been written.
+    SmallVector<std::string, 4> MetadataNames;
+    if (ImgPSInfo.EmitProgramMetadata) {
+      auto &ProgramMetadata =
+          PropSet[llvm::util::PropertySetRegistry::SYCL_PROGRAM_METADATA];
+
+      // Add reqd_work_group_size information to program metadata
+      for (const Function &Func : ResultModules[I]->functions()) {
+        std::vector<uint32_t> KernelReqdWorkGroupSize =
+            getKernelReqdWorkGroupSizeMetadata(Func);
+        if (KernelReqdWorkGroupSize.empty())
+          continue;
+        MetadataNames.push_back(Func.getName().str() + "@reqd_work_group_size");
```

Comment:
We write a string into the end of this vector and get this string on the next line. May we remove this vector and write string into ProgramMetadata directly?

Reply:
Ah, sorry, I didn't read comment before MetadataNames declration well enough. Nevermind.